hermes-frontend/src/main/java/pl/allegro/tech/hermes/frontend/publishing/handlers/TopicHandler.java

```diff
@@ -55,10 +56,14 @@ private void onTopicPresent(HttpServerExchange exchange, String messageId, Consu
         try {
             Optional<CachedTopic> topic = topicsCache.getTopic(qualifiedTopicName);
             if (topic.isPresent()) {
-                consumer.accept(topic.get());
-                return;
+                if (!topicsCache.isBlacklisted(qualifiedTopicName)) {
```

Comment:
Can we keep info about banned topic in CachedTopic object? If yes then read from another concurrent structure for every published message is unnecessary as we don't want to impact on frontend performance with this feature.